x/ccv/provider/migrations/v6/migrations.go

```diff
@@ -0,0 +1,33 @@
+package v6
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	providerkeeper "github.com/cosmos/interchain-security/v4/x/ccv/provider/keeper"
+)
+
+func MigrateMinPowerInTopN(ctx sdk.Context, providerKeeper providerkeeper.Keeper, stakingKeeper stakingkeeper.Keeper) {
+	// get all consumer chains
+	registeredConsumerChains := providerKeeper.GetAllRegisteredAndProposedChainIDs(ctx)
```

Comment:
Why also proposed chains? The minpower is computed only for registered chains. Also, `GetMinimumPowerInTopN` is never called for proposed chains.

Reply:
Right, doesn't seem necessary